evc-sumo/src/evc_connector.py

```diff
@@ -0,0 +1,147 @@
+from pyeos.virtual.factory import virtual_factory
+from pyeos.virtual import VirtualControllerOptions
+import json
+
+MAX_INTERSECTION_PHASES = 16
+
+class EvcConnector:
+
+    def tick(self):
+        """
+        Advance EVC controllers time via PyEOS
+        """
+        for harness in self.harness_list:
+            harness.tick(1)
+
+    def __init__(self, asc3app_path, evc_sumo_cfg_path):
+        """
+        Initialize EvcConnector object
+
+        Parameters:
+        - asc3app_path: The directory to asc3app which is provided by the Econolite team
+        type: string
+
+        - evc_sumo_cfg_path: The directory to evc_sumo_cfg.json file, default path is "resources/evc_sumo_cfg.json"
+        type: string
+        """
+        self.asc3app_path = asc3app_path
+        self.evc_sumo_cfg_path = evc_sumo_cfg_path
+        self.harness_list = []
+        self.controller_io_list = []
+
+    def detector_status_to_CIB(self):
+        """
+        Convert detector status to CIB
+        """
+        pass
+
+    def COB_to_traffic_light_status(self, controller_io, phase_id):
+        """
+        Convert COB status to string
+
+        Parameters:
+        - controller_io: The controller input/output (cib/cob)
+        type: pyeos.common.harness
+
+        - phase_id: The EVC traffic light phase ID, start from 1
+        type: int
+        """
+        ## This function converts the COB status for a given traffic light phase to a string representing its state (red, yellow, or green).
+        ## The controller input/output (cib/cob) is passed as an argument along with the phase ID.
+        ## The function returns a string representing the state of the traffic light phase.
+
+        ## is_cob_on(0) is to check if phase 1 is or not in green state
+        ## is_cob_on(16) is to check if phase 1 is or not in yellow state
+        ## is_cob_on(32) is to check if phase 1 is or not in red state
+        if controller_io.is_cob_on( (phase_id - 1) + (MAX_INTERSECTION_PHASES * 2) ):
+            return 'r'
+        elif controller_io.is_cob_on( (phase_id - 1) + (MAX_INTERSECTION_PHASES * 1) ):
+            return 'y'
+        elif controller_io.is_cob_on( (phase_id - 1) + (MAX_INTERSECTION_PHASES * 0) ):
+            return 'g'
+
+    def get_traffic_light_status_from_EVC(self, controller_io, phases):
+        """
+        Get traffic light status from EVC to SUMO state string
+
+        Parameters:
+        - controller_io: The controller input/output (cib/cob)
+        type: pyeos.common.harness
+
+        - phases: Phases information defined in evc_sumo_cfg.json file
+        type: dict
+
+        Returns:
+        - string: A string representing the current state of the traffic light. The possible characters are "r" for red, "y" for yellow, and "g" for green.
```

Comment:
I completely agree with your idea to use enumerated values instead of strings in the code. However, the only way to update the SUMO traffic light state is through the `traci.trafficlight.setRedYellowGreenState` function, which requires a string like 'rrggrrGG' to be passed as an argument.